test/runtests.jl

```diff
@@ -5,7 +5,7 @@ Pkg.develop(;
 
 using SparseConnectivityTracer
 using Documenter: Documenter, DocMeta
-using Test
+using Test, LinearAlgebra
```

Comment:
If this is for `I`, it looks like this should be in `test_gradient.jl` and `test_hessian.jl`?

Reply:
Maybe these horrible `Matrix(I(length(x)))` tests should not be written in the first place.

Reply:
The idea of the test is to have a dense matrix which has some zeros. It can happen even for less egregiously suboptimal functions than this one. And even this one may not be so bad, because for small sizes (as the people over in MPC are using), dense matrices are very likely to outperform `SparseMatrixCSC`. Of course they won't outperform `Diagonal` but that's an extreme example.

Reply:
Then just add two simple and informative scalar tests for `x * 0` and `0 * x` *in front* of these convoluted indirect ones.
Maybe something like `x^2 * 0` for Hessians.